File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        for (int i = 0; i < nums.size(); i++)
        {
            int num = nums[i];
            for (int j = 0; j < nums.size(); j++)
            {
                if (nums[j] == num && j != i)
                    return true;
            }
        }
        return false;
    }
};

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        left = 0
        right = len(nums) - 1
        minimum = float('inf')

        while left <= right:
            mid = (left + right) // 2
            if nums[mid] < minimum:
                minimum = nums[mid]

            if nums[mid] >= nums[right]:
                left = mid + 1
            else:
                right = mid - 1

        return minimum
            